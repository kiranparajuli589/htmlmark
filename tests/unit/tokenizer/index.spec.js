const path = require("path")
const { read } = require("../../../lib/file")
const tokenize = require("../../../lib/tokenizer/index")

describe("tokenization", () => {
  it("should tokenize a markdown content", () => {
    // eslint-disable-next-line no-undef
    const fileContent = read(path.join(__dirname, "..", "..", "fixtures", "markdown.md")).split("\n")
    expect(tokenize(fileContent)).toMatchSnapshot()
  })

})