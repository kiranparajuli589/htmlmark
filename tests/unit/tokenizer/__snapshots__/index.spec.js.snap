// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenization should tokenize a markdown content 1`] = `
Array [
  Object {
    "indent": "",
    "raw": "<!-- https://regex101.com/r/C6SxZH/1 -->",
    "tokens": Object {
      "value": "https://regex101.com/r/C6SxZH/1",
    },
    "type": "comment",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "indent": "",
    "level": 1,
    "raw": "# Heading One",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "Heading One",
      },
    ],
    "type": "heading",
  },
  Object {
    "indent": "",
    "level": 2,
    "raw": "## Heading Two",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "Heading Two",
      },
    ],
    "type": "heading",
  },
  Object {
    "indent": "",
    "level": 3,
    "raw": "### Heading Three",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "Heading Three",
      },
    ],
    "type": "heading",
  },
  Object {
    "indent": "",
    "level": 4,
    "raw": "#### Heading Four",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "Heading Four",
      },
    ],
    "type": "heading",
  },
  Object {
    "indent": "",
    "level": 5,
    "raw": "##### Heading Five",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "Heading Five",
      },
    ],
    "type": "heading",
  },
  Object {
    "indent": "",
    "level": 6,
    "raw": "###### Heading Six",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "Heading Six",
      },
    ],
    "type": "heading",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "raw": "a paragraph of words \`first code\` normal text here \`code body\` *first italics* here me crying *italic body* here me crying **first bolds** some normal again **bold body** [Kiran Parajuli](https://kiranparajuli.com.np) ~~strikes body~~ here some normal again at the last",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "a paragraph of words ",
      },
      Object {
        "type": "code",
        "value": "first code",
      },
      Object {
        "type": "text",
        "value": " normal text here ",
      },
      Object {
        "type": "code",
        "value": "code body",
      },
      Object {
        "type": "text",
        "value": " ",
      },
      Object {
        "type": "italic",
        "value": "first italics",
      },
      Object {
        "type": "text",
        "value": " here me crying ",
      },
      Object {
        "type": "italic",
        "value": "italic body",
      },
      Object {
        "type": "text",
        "value": " here me crying ",
      },
      Object {
        "type": "bold",
        "value": "first bolds",
      },
      Object {
        "type": "text",
        "value": " some normal again ",
      },
      Object {
        "type": "bold",
        "value": "bold body",
      },
      Object {
        "type": "text",
        "value": " ",
      },
      Object {
        "href": "https://kiranparajuli.com.np",
        "type": "link",
        "value": "Kiran Parajuli",
      },
      Object {
        "type": "text",
        "value": " ",
      },
      Object {
        "type": "strike-through",
        "value": "strikes body",
      },
      Object {
        "type": "text",
        "value": " here some normal again at the last",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "depth": 0,
    "indent": "",
    "raw": "> some quote",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "some quote",
      },
    ],
    "type": "quote",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "indent": "",
    "raw": "- list item 1",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "list item 1",
      },
    ],
    "type": "list-item",
  },
  Object {
    "indent": "",
    "raw": "- list item 2",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "list item 2",
      },
    ],
    "type": "list-item",
  },
  Object {
    "indent": "",
    "raw": "- [ ] checkbox empty",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "checkbox empty",
      },
    ],
    "type": "check-box",
  },
  Object {
    "indent": "",
    "raw": "- [x] checkbox checked",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "checkbox checked",
      },
    ],
    "type": "check-box-checked",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "depth": 0,
    "indent": "",
    "raw": "> quote 1",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "quote 1",
      },
    ],
    "type": "quote",
  },
  Object {
    "depth": 1,
    "indent": "",
    "raw": "> > nested quote 2 [quote-link](quote-link-url)",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "nested quote 2 ",
      },
      Object {
        "href": "quote-link-url",
        "type": "link",
        "value": "quote-link",
      },
    ],
    "type": "quote",
  },
  Object {
    "depth": 2,
    "indent": "",
    "raw": "> > > nested quote 3 with *italics* and **bolds**",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "nested quote 3 with ",
      },
      Object {
        "type": "italic",
        "value": "italics",
      },
      Object {
        "type": "text",
        "value": " and ",
      },
      Object {
        "type": "bold",
        "value": "bolds",
      },
    ],
    "type": "quote",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "language": "js",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "const a = 1;
const b = 1;
const c = a + b;

if (c === 2) {
  console.log('success');
} else {
  console.log('failure');
}",
      },
    ],
    "type": "code-block",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "language": "js",
    "tokens": Array [
      Object {
        "type": "text",
        "value": null,
      },
    ],
    "type": "code-block",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "raw": "some piece of normal paragraph",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "some piece of normal paragraph",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "language": null,
    "tokens": Array [
      Object {
        "type": "text",
        "value": "another code block is here",
      },
    ],
    "type": "code-block",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "indent": "",
    "raw": "![alt text](https://kiranparajuli.com.np/images/kiran.jpg)",
    "tokens": Object {
      "alt": "alt text",
      "url": "https://kiranparajuli.com.np/images/kiran.jpg",
    },
    "type": "image",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "raw": "some random text",
    "tokens": Array [
      Object {
        "type": "text",
        "value": "some random text",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "type": "new-line",
  },
]
`;
